core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -734,7 +735,7 @@ public SqlNode toSql(@Nullable RexProgram program, RexNode rex) {
           accesses.offerLast((RexFieldAccess) referencedExpr);
           referencedExpr = ((RexFieldAccess) referencedExpr).getReferenceExpr();
         }
-        SqlFieldAccess sqlFieldAccess = new SqlFieldAccess(POS);
+        List<SqlNode> names = new ArrayList<>();
```

Comment:
SqlFieldAccess was already a list of SqlNodes. Why did we feel a need to change this?

Reply:
this is done to build either a SqlObjectAccess or SqlFieldAccess. SqlObjectAccess has a special syntax  `(object_column).field`. SqlFieldAccess is currently being used for struct fields in big query and does not need any such special syntax and can be replaced by SqlIdentifier instead.